File: 1094.py
length = int(input())
nums = list(map(int, input().split()))

def minMoves(n, arr):
    res, prev = 0, arr[0]
    for i in range(1, n):
        if arr[i] < prev:
            res += prev - arr[i]
        prev = max(arr[i], prev)
    return res

print(minMoves(length, nums))

File: 1094.cpp
#include <iostream>
using namespace std;

int main() {
	int n, x, curr = 0;
	long long res = 0;
	cin >> n;

	for (int i = 0; i < n; i++) {
		cin >> x;
		curr = max(curr, x);
		res += max(curr - x, 0);
	}

	cout << res;
}
